lib/puppet/pops/evaluator/collectors/abstract_collector.rb

```diff
@@ -45,8 +45,10 @@ def evaluate
     return false if objects.empty?
 
     if @overrides and !objects.empty?
-      overrides[:source].meta_def(:child_of?) do |klass|
-        true
+      overrides[:source].instance_eval do
+        def child_of?(klass)
```

Comment:
fwiw, creating a toggle on the Puppet::Resource::Type class that child_of? checks first and then setting that here would get us away from defining methods on an instance at runtime completely.